### ISSUE TITLE
Frequent System Passord Prompts when override external modifications enabled (8.3 on yosemite)

### ISSUE THREAD

**Issue by drdamour** [2ndalpha/gasmask#85]
Ever since yosemite i get the system password prompt when override external modifications is enabled.

I get it when i wake from sleep, or come from a lock screen. I also get it when connecting or disconnecting junos pulse secure vpn client.  this makes me think ti's every time i'm establishing a new network connection.  I don't seem to get it when hopping wifi networks..but that may be cause my system pass is in the grace period.

Seems like something is tripping the external modification detector.

**Comment by trevershick** [comment]
It's due to the pulse secure vpn client. I use the same client. When it disconnects even briefly it restores /etc/hosts with a copy it makes when it connects.  Thus it triggers the 'restore' with gas mask.
It's annoying but that's how it works.  You can always just disable the restoration feature in the gas mask preferences.

**Comment by drdamour** [comment]
could gas mask do a diff on the file to see if there's really a change before triggering external modification routine?

**Comment by trevershick** [comment]
I suppose that's possible. It being the holidays i'm not sure when i'll get to it.
Although i'd like to see this feature myself.

**Comment by drdamour** [comment]
might related to #84 or #58

**Comment by trevershick** [comment]
Yeah. the system password prompt actually fixed a defect that was in 8.1.  8.1 wasn't properly restoring the hosts file becuase the permissions on hosts required root to modify the file.  It's unfortunate, but it's the way it is.  However, i think with your suggestion it may minimize the 'noise' for these users.

**Comment by robhuzzey** [comment]
I'm having a similar issue but for me it also freezes after I have entered the password & refuses to accept it.

A machine restart seems to be the only fix :/

**Comment by mrjcleaver** [comment]
I too am using Pulse Secure. It'd be nice to be able to automatically accept a legitimate change. It seems to install and remove the VPN's ssl address IP address in a demarked block.

**Comment by MarsVard** [comment]
Is there any way to know which application is modifying the hosts file causing gasmask to restore it?

**Comment by zupodaniel** [comment]
I'm having the same problem with macOS 10.12.x and parallels running

**Comment by kfattic** [comment]
Yes, I have this problem with Parallels as well.  Annoying, but other than running it as root (sudo /Applications/Gas\ Mask.app/Contents/MacOS/Gas\ Mask) not sure what else to do.
